resources/2025-11-11-secrets-redactdata.md

```diff
@@ -0,0 +1,477 @@
+# Redacting Sensitive Data for Radius.Security/secrets
+
+* **Author**: Lakshmi Javadekar (@lakshmimsft)
+
+## Overview
+
+The `Radius.Security/secrets` resource type, defined in the [resource-types-contrib](https://github.com/radius-project/resource-types-contrib) repository, enables users to deploy sensitive data such as tokens, passwords, keys, and certificates to various secret backends (Azure Key Vault, HashiCorp Vault, Kubernetes secrets, etc.). This resource is handled by dynamic-rp, a type-agnostic processor providing CRUD operations for user-defined types.
+
+Currently, dynamic-rp stores all resource properties, including sensitive `data` fields, in the Radius database. This design proposes a mechanism to redact (nullify) sensitive data after the recipe successfully deploys the secrets to their designated backend, ensuring that secret data does not persist in the Radius database.
+
+## Terms and definitions
+
+| Term | Definition |
+|------|------------|
+| **Dynamic-RP** | Type-agnostic resource provider that handles CRUD operations for user-defined types without prior knowledge of their schemas |
+| **Recipe** | Deployment template (Bicep/Terraform) that provisions infrastructure resources |
+| **Recipe-based deployment** | Asynchronous deployment flow where recipes execute to create actual infrastructure |
+| **Radius.Security/secrets** | New resource type for managing secrets across multiple backends (recipe-based, multi-backend) |
+
+## Objectives
+
+> **Issue Reference:** Implementation of sensitive data handling for Radius.Security/secrets resource type to prevent long-term storage of plaintext secrets in Radius database. ref: https://github.com/radius-project/radius/issues/10421
+
+### Goals
+
+1. **Prevent persistent storage**: Ensure sensitive data is nullified from Radius database after successful recipe deployment
+2. **Minimal impact to dynamic-rp core**: Implement redaction as an extensible processor hook without requiring dynamic-rp to understand resource schemas.
+3. **Enable recipe-based secret deployment**: Allow recipes to access sensitive data during execution to deploy secrets to various backends (Azure Key Vault, HashiCorp Vault, K8s, etc.)
+4. **Support multi-backend flexibility**: Enable the same implementation to work with any recipe backend
+
+
+### Non goals
+
+1. **Prevent all temporary storage**: The design accepts that sensitive data will temporarily exist in the database during recipe execution (seconds to minutes)
```

Comment:
Curious why this temporary step is needed? I would move away from storing it in database altogether.